src/config.js

```diff
@@ -0,0 +1,64 @@
+/* @flow */
+import path from 'path';
+
+import requireUncached from 'require-uncached';
+
+import {createLogger} from './util/logger';
+import {UsageError} from './errors';
+
+const log = createLogger(__filename);
+
+type ApplyConfigToArgvParams = {|
+  argv: Object,
+  configObject: Object,
+  defaultValues: Object,
+|};
+
+type ParseConfigParams = {|
+  sourceDir: string,
+  configFileName: string,
+|};
+
+export function applyConfigToArgv({
+  argv,
+  configObject,
+  defaultValues = {},
+}: ApplyConfigToArgvParams): Object {
+  const newArgv = {...argv};
+  for (const option in configObject) {
+    const wasValueSetOnCLI = argv[option] && argv[option]
+                            !== defaultValues[option];
+    if (wasValueSetOnCLI) {
+      continue;
+    }
+    if (!argv.hasOwnProperty(option) || defaultValues[option]) {
+      newArgv[option] = configObject[option];
+      log.debug(`Favoring configuration: ${option}=${configObject[option]}` +
+      `over CLI: ${option}=${argv[option]}`);
+    }
+  }
+  return newArgv;
+}
+
+export function parseConfig({
+  sourceDir,
+  configFileName,
+}: ParseConfigParams): Object {
+  let configObject = {};
+  if (configFileName) {
+    const configFilePath = path.join(`/${sourceDir}`, configFileName);
```

Comment:
If I remove this, there will be no default place to look. Even the existing test will fail. How will I pass `tmpDir.path()` to `parseConfig()`?